apps/prairielearn/src/lib/db-types.ts

```diff
@@ -84,8 +84,269 @@ export const DateFromISOString = z
+  last_submission_score_hist: z.array(z.number()).nullable(),
+  last_submission_score_variance: z.number().nullable(),
+  manual_rubric_id: IdSchema.nullable(),
+  max_auto_points: z.number().nullable(),
+  max_manual_points: z.number().nullable(),
+  max_points: z.number().nullable(),
+  max_submission_score_hist: z.array(z.number()).nullable(),
+  max_submission_score_variance: z.number().nullable(),
+  mean_question_score: z.number().nullable(),
+  median_question_score: z.number().nullable(),
+  number: z.number().nullable(),
+  number_in_alternative_group: z.number().nullable(),
+  number_submissions_hist: z.array(z.number()).nullable(),
+  number_submissions_variance: z.number().nullable(),
+  points_list: z.array(z.number()).nullable(),
+  question_id: IdSchema,
+  question_score_variance: z.number().nullable(),
+  quintile_question_scores: z.array(z.number()).nullable(),
+  some_nonzero_submission_perc: z.number().nullable(),
+  some_perfect_submission_perc: z.number().nullable(),
+  some_submission_perc: z.number().nullable(),
+  submission_score_array_averages: z.array(z.number()).nullable(),
+  submission_score_array_variances: z.array(z.number()).nullable(),
+  tries_per_variant: z.number().nullable(),
+});
+export type AssessmentQuestion = z.infer<typeof AssessmentQuestionSchema>;
+
+export const AssessmentQuestionRolePermissionsSchema = z.object({
+  assessment_question_id: IdSchema,
+  group_role_id: IdSchema,
+  can_submit: z.boolean().nullable(),
+  can_view: z.boolean().nullable(),
+});
+export type AssessmentQuestionRolePermissions = z.infer<
+  typeof AssessmentQuestionRolePermissionsSchema
+>;
+
+export const AssessmentSetSchema = z.object({
+  abbreviation: z.string(),
+  color: z.string(),
+  course_id: IdSchema,
+  heading: z.string(),
+  id: IdSchema,
+  implicit: z.boolean(),
+  name: z.string(),
+  number: z.number(),
+});
+export type AssessmentSet = z.infer<typeof AssessmentSetSchema>;
+
+export const AuditLogSchema = z.object({
+  action: z.string().nullable(),
+  authn_user_id: IdSchema.nullable(),
+  column_name: z.string().nullable(),
+  course_id: IdSchema.nullable(),
+  course_instance_id: IdSchema.nullable(),
+  date: DateFromISOString.nullable(),
+  group_id: IdSchema.nullable(),
+  id: IdSchema,
+  institution_id: IdSchema.nullable(),
+  new_state: z.any(),
+  old_state: z.any(),
+  parameters: z.any(),
+  row_id: IdSchema.nullable(),
+  table_name: z.string().nullable(),
+  user_id: IdSchema.nullable(),
+});
+export type AuditLog = z.infer<typeof AuditLogSchema>;
+
+export const AuthnProviderSchema = z.object({
+  id: IdSchema,
+  name: z.string().nullable(),
+});
+export type AuthnProvider = z.infer<typeof AuthnProviderSchema>;
+
+export const ClientFingerprintSchema = z.object({
+  id: IdSchema,
+  user_id: IdSchema,
+  user_session_id: IdSchema,
+  ip_address: z.string(),
+  user_agent: z.string().nullable(),
+  accept_language: z.string().nullable(),
+  created_at: DateFromISOString,
```

Comment:
Do you want to enforce column alphabetical order as well?

Reply:
Done in d88202c.